Add method to retrieve body of closure in stable-mir

b-naber opened the pull request:
fixes https://github.com/rust-lang/project-stable-mir/issues/85

r? @celinval

b-naber committed:
add `body` to `ClosureDef`

celinval commented on compiler/rustc_smir/src/stable_mir/ty.rs:
```diff
@@ -742,6 +742,12 @@ crate_def! {
 }
 
+impl ClosureDef {
+    pub fn body(&self) -> Option<Body> {
```
Can you please add documentation?

celinval reviewed (commented):
Awesome! If it's not too much work, can you please add a test for a closure that takes a generic argument?

b-naber committed:
add doc comment and a test with a generic closure

b-naber commented:
Sorry for addressing your comments so late.

I have added the test with the generic closure and a doc comment (though I'm not sure it offers much information).

bors commented:
:pushpin: Commit 8604e58942924cde5e96d223130c6d6200677af7 has been approved by `celinval`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=celinval 8604e58942924cde5e96d223130c6d6200677af7 -->
<!-- homu: {"type":"Approved","sha":"8604e58942924cde5e96d223130c6d6200677af7","approver":"celinval","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
